rfd/0163-vnet.md

```diff
@@ -0,0 +1,792 @@
+
+#### Managing VNet
+
+<img width="871" alt="VNet in Connections" src="assets/0163-vnet-in-connections.png">
+
+VNet has a persistent place in the UI in the connection list in top left. From there, the user can
+see VNet status for the current root cluster and quickly stop VNet for all root clusters, similar to
+how they can disconnect existing database, SSH and Kube connections.
+
+Clicking on the VNet item opens a dedicated VNet tab. Data in this tab is scoped to the root cluster
+of the current workspace. The user can see the status of VNet for the root cluster, along with an
+error message if there was a problem with setting up VNet. The tab displays the DNS zones that are
+currently being proxied and recent connections.
+
+A green indicator means that a tunnel is currently open. The absence of an indicator means the
+tunnel was opened but has since been closed due to inactivity. An error indicator means that a
+connection for the given app could not be established. Clicking on the row with the error shows the
+full error message.
+
+<img width="871" alt="VNet tab" src="assets/0163-vnet-tab.png">
+
+#### Lifecycle
+
+The first version of VNet integration in Connect behaves similarly to Connect My Computer. After the
+user starts VNet and until they explicitly stop it, VNet starts automatically when launching the
+app. It prompts for a password for root privileges through osascript on each start.
+
+If VNet is running and the user logs in to another cluster, Connect prompts for credentials again
+and configures VNet for that cluster. If this attempt fails, VNet continues to work but the
+indicator on the connection list icon shows a red exclamation mark. Upon opening the VNet tab from
+the connection list, the user can learn what the error was and either retry the attempt or ignore
+the error.
+
+Unfortunately, [Touch ID prompts are reserved for Apple
+clients](https://forums.developer.apple.com/forums/thread/701118?answerId=705661022#705661022). We
+assume that typically the user is going to start VNet and then just leave it running, somewhat
+negating the downside of having to enter credentials on each start.
+
+##### Manual start and stop
+
+Starting VNet by clicking the item in the connection list in the top left starts VNet for all root
+clusters and turns on autostart for all of them. Stopping VNet from the connection list stops it for
+all root clusters and turns off autostart for all of them.
+
+If VNet is not running, starting VNet from the VNet tab starts it only for the root cluster of the
+current workspace. Likewise, stopping it from the VNet tab stops it only for the current root
+cluster and turns off autostart just for this cluster.
+
+This behavior aligns with a couple of ideas:
+
+1) The connection list is a UI element that is not scoped to any workspace, unlike tabs which are
+always scoped to a specific workspace.
+2) The user can turn off VNet for a specific workspace if they do not wish to use it in that
+workspace.
```

Comment:
Would it be simpler to just have it global on/off? There's fewer options for the user and we wouldn't need to communicate from connect all the way to the root process which domains we should configure DNS for, the root process could figure it out itself by just looking at all the profiles in TSH_HOME

Reply:
Addressed in https://github.com/gravitational/teleport/pull/37834#discussion_r1511469472.